include/kllvm/binary/serializer.h

```diff
@@ -141,6 +141,34 @@ void serializer::emit(T val) {
 
 void emit_kore_rich_header(std::ostream &os, kore_definition *definition);
 
+class proof_trace_writer {
+public:
+  virtual ~proof_trace_writer() = default;
+  virtual void write(void const *ptr, size_t len) = 0;
+  virtual void write_string(char const *str, size_t len) = 0;
+  virtual void write_string(char const *str) = 0;
```

Comment:
What's the intended difference between this overload and `write_c_string`?

Reply:
`write_c_string` is supposed to add the `\0` at the end, while the `write_string` overloads do not add a `\0`. I understand that the naming is not ideal and I am open to suggestions.

Reply:
`write_string` and `write_terminated_string` maybe? Also worth adding a comment, perhaps

Reply:
I just pushed a commit where I have done the renaming and added comments as you recommended.